Created DropDownWithLabel component and Fixed dropdown styling to match Figma

kea-roy opened the pull request:
## Summary <!-- Required -->

Created DropDownWithLabel component and revamped all pages with dropdown component to avoid repetitive code. Fixed dropdown styling to better match Figma. This pull request is a continuation of pull requests #352 and #353, specifically to resolve https://github.com/cornell-dti/cu-apts/pull/353#pullrequestreview-1956440909 

## Changes
- [x] Created DropDownWithLabel component, with label styling and mobile support 
- [x] Replaced DropDown components with DropDownWithLabel components on all pages.
- [x] Adjusted dropdown margins and font sizes to match Figma
- [x] Changed dropdown arrow icon to match Figma
- [x] Added dropdown arrow icon animation when the menu is opened 

### Test Plan <!-- Required -->

Checked every webpage that had a dropdown to make sure the styling was correct on both desktop and mobile screens.
Screenshots of the webpage on desktop and mobile screens are shown below

<img width="400" src="https://github.com/cornell-dti/cu-apts/assets/146872846/69474387-2b4b-41cd-8055-6b16562e6bd2">
<img width="400" src="https://github.com/cornell-dti/cu-apts/assets/146872846/56562376-d096-4b18-a629-e47647303578">
<img width="400" src="https://github.com/cornell-dti/cu-apts/assets/146872846/6d1b4452-fedc-49b8-8703-3b5c52530308">
<img width="400" src="https://github.com/cornell-dti/cu-apts/assets/146872846/af36fa5c-0416-4627-abf1-ef17f72556e7">
<img width="1027" alt="Screenshot 2024-04-09 at 18 00 32" src="https://github.com/cornell-dti/cu-apts/assets/146872846/026d7544-ca59-4e86-8db9-445248f70bdd">
<img width="1027" alt="Screenshot 2024-04-09 at 18 03 38" src="https://github.com/cornell-dti/cu-apts/assets/146872846/d75a7558-040e-4fc9-be95-84c364bf7157">
<img width="1027" alt="Screenshot 2024-04-09 at 18 04 04" src="https://github.com/cornell-dti/cu-apts/assets/146872846/79e84c5f-99d5-41e1-9171-6429eee5b4a1">
<img width="1027" alt="Screenshot 2024-04-09 at 18 04 40" src="https://github.com/cornell-dti/cu-apts/assets/146872846/58c0d15e-de02-4b69-b2c6-536ce36fe0bf">
<img width="1027" alt="Screenshot 2024-04-09 at 18 00 43" src="https://github.com/cornell-dti/cu-apts/assets/146872846/e75a5a75-6b6f-4924-9a10-4e4efa150039">

kea-roy commented:
## Commit Changes

1. Improved documentation for DropDownWithLabel

CasperL1218 commented on frontend/src/components/utils/DropDown.tsx:
```diff
@@ -27,8 +27,22 @@
+  return (
+    <div style={{ paddingLeft: '10px', transform: direction ? 'rotate(180deg)' : 'none' }}>
+      <svg
+        xmlns="http://www.w3.org/2000/svg"
```
I am not sure if this is necessary, but I think creating a local file for the icon/svg in the asset folder and referencing the icon from the asset would be:
1) More flexible for future modification
2) Easier for inspection 
3) Conforming to the general code pattern

kea-roy replied:
Thanks for pointing this out! I've updated my code to fix this code format inconsistency in commit 89e93ba.

CasperL1218 reviewed (changes requested):
Your implementation and refinement of the overall dropdown mechanism are fully robust and complete after multiple iterations. The refactoring of the dropDownLabel component makes the code even more organized, which is really helpful. 

There are two small things that are not directly related to the dropdown mechanism, but you might want to look into:
(1). The referencing to the SVG file arrow icon is using a "http" path. I believe it could be replaced with creating a SVG file for the icon in the asset folder for reference to both offer an easier inspection and align with the code base practices.
(2). The import of {Review} in backend/scripts/add_price_and_bedroom_attributes.ts might not be necessary and can be removed unless you are using it in later implementation.

kea-roy commented:
## Commit Changes
1. Added animation for dropdown
2. Adjusted code formatting for DropDown.tsx
3. Moved dropdown arrow svg into a separate file and linked it via an img element.
